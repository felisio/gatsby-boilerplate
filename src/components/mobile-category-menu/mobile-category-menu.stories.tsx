import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { INITIAL_VIEWPORTS } from '@storybook/addon-viewport';

import MobileCategoryMenu, { MobileCategoryMenuProps } from './mobile-category-menu';
import categoryList from '../../../__mocks__/category-menu';
import { WpCategory } from '../../../graphql-types';

export default {
    title: 'Components/MobileCategoryMenu',
    component: MobileCategoryMenu,
    decorators: [
        StoryContainer => (
            <div>
                <StoryContainer />
            </div>
        ),
    ],
    parameters: {
        viewport: {
            viewports: INITIAL_VIEWPORTS,
            defaultViewport: 'iphone6',
        },
        chromatic: { viewports: [375] }, // width of iPhone 6
    },
} as Meta;

const Template: Story<MobileCategoryMenuProps> = args => <MobileCategoryMenu {...args} />;

export const Primary = Template.bind({});
Primary.args = {
    title: 'Select Category',
    categoryList: categoryList as WpCategory[],
};

export const WithSearchValue = Template.bind({});
WithSearchValue.args = {
    title: 'Select Category',
    categoryList: categoryList as WpCategory[],
    searchValue: 'Quantic search',
};

export const WithActiveLink = Template.bind({});
WithActiveLink.args = {
    title: 'Select Category',
    categoryList: categoryList as WpCategory[],
    urlActive: '/category/entrepreneurship/',
};
