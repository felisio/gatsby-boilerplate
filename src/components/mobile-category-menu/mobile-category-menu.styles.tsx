import styled, { css } from 'styled-components';
import { Container } from '../../styles/Container';
import { mediaQuery } from '../../styles/mediaQuery';

const StyledMobileCategoryHeader = styled(Container)`
    .header {
        display: flex;
        justify-content: flex-end;
        height: 22px;
        padding: 0 0.9rem;
        margin-bottom: 0.8rem;
    }

    .divider {
        width: 100%;
        height: 1px;
        background-image: linear-gradient(
            to right,
            rgba(155, 168, 187, 0),
            rgba(155, 168, 187, 0.3) 49%,
            rgba(155, 168, 187, 0)
        );
    }
`;

const StyledMobileCategoryStick = styled.div<{ hasTopBanner: boolean; showStick: boolean; stick: boolean }>`
    --menu-margin-top: 50px;

    position: fixed;
    top: ${props => (props.hasTopBanner ? 'var(--menu-margin-top)' : '0')};
    z-index: 5;
    display: ${props => (props.showStick ? 'block' : 'none')};

    width: 100%;
    padding: 2.3rem 0 2rem;
    background-color: var(--white);
    box-shadow: 0 8px 12px 0 rgba(0, 0, 0, 0.1);
    transition: transform 0.6s ease-in-out;
    transform: ${props => (props.stick ? 'translateY(0)' : 'translateY(-130px)')};
    will-change: transform;

    ${mediaQuery.upSm} {
        --menu-margin-top: 30px;
    }

    ${mediaQuery.upMd} {
        display: none;
    }
`;

const StyledMobileCategoryMenu = styled.div<{ hasTopBanner: boolean }>`
    ${mediaQuery.upMd} {
        display: none;
    }

    display: block;
    width: 100%;
    padding: 2.3rem 0 2rem;

    &.stick {
        position: fixed;
        top: ${props => (props.hasTopBanner ? '50px' : '0')};
        background-color: var(--white);
        box-shadow: 0 8px 12px 0 rgba(0, 0, 0, 0.1);
        transition: all 0.5s ease-in;
        animation-name: topDown;
        animation-duration: 0.5s;
        animation-timing-function: ease-in;
        will-change: transform;

        @keyframes topDown {
            0% {
                top: -40px;
            }

            100% {
                top: ${props => (props.hasTopBanner ? '50px' : '0')};
            }
        }
    }
`;

const StyledCategorySelect = styled.div<{ isSearchOpen: boolean; isDropdownOpen: boolean }>`
    display: ${props => (props.isSearchOpen ? 'none' : 'flex')};
    align-items: center;
    justify-content: space-between;
    width: 100%;
    padding-right: 0.1rem;
    cursor: pointer;
    animation: ContentAnimation 0.8s forwards;

    @keyframes ContentAnimation {
        0% {
            width: 0;
            opacity: 0;
        }

        100% {
            width: 100%;
            opacity: 1;
        }
    }

    .title {
        max-width: 100%;
        overflow: hidden;
        font-size: 1.8rem;
        font-weight: normal;
        font-stretch: normal;
        line-height: normal;
        color: var(${props => (props.isDropdownOpen ? '--fiord' : '--lynch')});
        text-overflow: ellipsis;
        letter-spacing: normal;
        white-space: nowrap;
    }

    img {
        transition: all 0.5s ease-in-out;
        transform: rotate(${props => (props.isDropdownOpen ? '-180deg' : '0deg')});
    }
`;

const StyledCategorySelectSearch = styled.div<{ isOpen: boolean }>`
    position: relative;
    display: flex;
    width: 50px;
    transition-duration: 1s;
    transition-property: width;

    &.open {
        width: 100%;
    }

    button {
        position: relative;
        cursor: pointer;
        background-color: transparent;
        border: none;
        outline: none;

        &:focus {
            border: none;
            outline: none;
        }
    }

    .search-button {
        padding: ${props => (props.isOpen ? '0' : '0 0 0 1.6rem')};
        transition: 0.4s ease-in-out;
    }

    .close-button {
        position: absolute;
        top: 0;
        right: 0;
        opacity: 1;
        animation-name: fadeInVisibility;
        animation-duration: 1s;
        animation-timing-function: ease-in;

        @keyframes fadeInVisibility {
            0% {
                opacity: 0;
            }

            100% {
                opacity: 1;
            }
        }
    }

    input[type='search'] {
        display: block;
        width: 80%;
        padding: 0 0 0 0.9rem;
        font-size: 1.8rem;
        color: var(--fiord);
        background-color: transparent;
        border: none;
        caret-color: var(--wild-watermelon);

        /* clears the ‘X’ from Chrome */
        &::-webkit-search-decoration,
        &::-webkit-search-cancel-button,
        &::-webkit-search-results-button,
        &::-webkit-search-results-decoration {
            display: none;
        }

        &:focus {
            border: none;
            outline: none;
        }
    }
`;

const sharedLinkStyles = css`
    a {
        display: block;
    }

    a,
    button {
        padding: 0.4rem 0;
        font-size: 1.5rem;
        font-weight: 500;
        line-height: 2;
        color: var(--shuttle-grey);
        text-decoration: none;
        letter-spacing: 0.23px;
        transition: 0.2s ease;
    }

    &.active {
        a {
            color: var(--wild-watermelon);
        }
    }
`;

const StyledMobileCategoryBodyHeader = styled.header`
    position: fixed;
    top: 0;
    left: 0;
    z-index: 20;
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 100%;
    height: 60px;
    padding: 1rem 1.5rem 1.3rem 2.4rem;
    overflow-y: hidden;
    background-color: var(--white);

    &.border-bottom {
        &::after {
            position: absolute;
            bottom: 0;
            left: 0;
            width: 100%;
            height: 1px;
            content: '';
            background-color: rgba(121, 137, 159, 0.8);
            opacity: 0.15;
        }
    }

    h1 {
        padding: 0;
        margin: 0;
        font-size: 1.8rem;
        font-weight: 500;
        font-stretch: normal;
        color: var(--wild-watermelon);
        letter-spacing: normal;
    }

    button {
        position: relative;
        cursor: pointer;
        background-color: transparent;
        border: none;
        outline: none;

        &:focus {
            border: none;
            outline: none;
        }
    }
`;

const StyledCategoryDropdownShape = styled.div<{ isDropdownOpen: boolean }>`
    position: fixed;
    top: 60px;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: 19;
    display: block;
    height: 100vh;
    overflow-x: hidden;
    overflow-y: auto;
    pointer-events: ${props => (props.isDropdownOpen ? 'auto' : 'none')};
    background-color: var(--white);
    background-clip: padding-box;
    opacity: ${props => (props.isDropdownOpen ? 1 : 0)};
    transition: 0.2s ease-in-out;
    transform: translateY(${props => (props.isDropdownOpen ? 0 : '-8px')});

    --padding-bottom: calc(100% - 22em);

    .content {
        padding: 0 2.4rem var(--padding-bottom) 2.4rem;

        @media (orientation: landscape) {
            --padding-bottom: calc(50% - 22em);
        }
    }

    ul {
        padding: 0;
        margin: 0;
        overflow: auto;
        list-style: none;

        li {
            &.submenu {
                --border-color: rgba(121, 137, 159, 0.1);

                border-top: 1px solid var(--border-color);
                border-bottom: 1px solid var(--border-color);

                &:last-child {
                    border-bottom: none;
                }

                &:first-child {
                    border-top: none;
                }
            }

            ${sharedLinkStyles}
        }
    }
`;

const StyledMobileCategorySubMenu = styled.li<{ isOpen: boolean }>`
    --border-color: rgba(121, 137, 159, 0.1);

    margin-top: 0.4rem;
    border-top: 1px solid var(--border-color);
    border-bottom: 1px solid var(--border-color);

    &:last-child {
        border-bottom: none;
    }

    &:first-child {
        border-top: none;
    }

    button {
        display: flex;
        justify-content: space-between;
        width: 100%;
        padding: 0;
        margin: 0;
        text-align: start;
        cursor: pointer;
        background-color: transparent;
        border: none;
        outline: none;

        &:focus {
            border: none;
            outline: none;
        }

        img {
            position: relative;
            top: 9px;
            transition: all 0.5s ease-in-out;
            transform: rotate(${props => (props.isOpen ? '-180deg' : '0deg')});
        }
    }

    ul {
        display: ${props => (props.isOpen ? 'initial' : 'none')};
        width: 100%;
        opacity: ${props => (props.isOpen ? 1 : 0)};
        transition: 0.2s ease-in-out;
        transform: translateY(${props => (props.isOpen ? 0 : '-8px')});

        li {
            padding-left: 1.6rem;

            ${sharedLinkStyles}
        }
    }
`;

export {
    StyledMobileCategoryMenu,
    StyledMobileCategoryStick,
    StyledMobileCategoryHeader,
    StyledCategorySelect,
    StyledCategorySelectSearch,
    StyledCategoryDropdownShape,
    StyledMobileCategorySubMenu,
    StyledMobileCategoryBodyHeader,
};
