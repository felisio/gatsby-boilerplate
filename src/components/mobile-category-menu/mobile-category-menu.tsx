import React, { useEffect, useLayoutEffect, useRef, useState } from 'react';
import { navigate } from 'gatsby';
import {
    StyledMobileCategoryMenu,
    StyledMobileCategoryStick,
    StyledCategorySelect,
    StyledCategorySelectSearch,
    StyledCategoryDropdownShape,
    StyledMobileCategorySubMenu,
    StyledMobileCategoryHeader,
    StyledMobileCategoryBodyHeader,
} from './mobile-category-menu.styles';
import { Maybe, WpCategory } from '../../../graphql-types';
import Icon from '../icon/icon';
import { GlobalIconName } from '../icon/iconData';
import { ComponentState } from '../../utils/componentState';
import ListItemAnchor from '../list-item-anchor/list-item-anchor';
import useStick from '../../hooks/use-stick';

export type MobileCategoryMenuProps = {
    title?: string;
    categoryList?: Maybe<WpCategory>[];
    searchUrl?: string;
    searchValue?: string;
    urlActive?: string;
    hasTopBanner?: boolean;
    stick?: boolean;
};

export type MobileCategoryHeaderProps = {
    title?: string;
    searchUrl?: string;
    searchValue?: string;
    isDropdownOpen?: boolean;
    handleDropdown: (state: boolean) => void;
};

export type MobileCategoryBodyProps = {
    title?: string;
    categoryList?: Maybe<WpCategory>[];
    urlActive?: string;
    isDropdownOpen?: boolean;
    handleCloseDropdown: () => void;
};

function MobileCategorySubMenu({ title, categoryList, urlActive = '' }: MobileCategoryMenuProps) {
    const [isOpen, setIsOpen] = useState<boolean>(false);
    const stateIcon = isOpen ? ComponentState.selected : ComponentState.normal;
    return (
        <StyledMobileCategorySubMenu isOpen={isOpen}>
            <button type="button" onClick={() => setIsOpen(!isOpen)}>
                {title}
                <Icon iconName={GlobalIconName.downChevron16} state={stateIcon} />
            </button>
            <ul>
                {categoryList?.map(item => {
                    const isLinkActive = urlActive === item?.uri;
                    return (
                        <ListItemAnchor
                            key={item?.id}
                            title={item?.name!}
                            url={item?.uri!}
                            className={isLinkActive ? 'active' : ''}
                        />
                    );
                })}
            </ul>
        </StyledMobileCategorySubMenu>
    );
}

export function MobileCategoryHeader({
    title,
    searchUrl = 'search',
    searchValue = '',
    isDropdownOpen = false,
    handleDropdown,
}: MobileCategoryHeaderProps) {
    const inputRef = useRef<HTMLInputElement>(null);
    const [isSearchOpen, setIsSearchOpen] = useState<boolean>(false);
    const [inputValue, setInputValue] = useState<string>(searchValue);
    const [inputFocus, setInputFocus] = useState<boolean>(false);

    useEffect(() => {
        setIsSearchOpen(Boolean(searchValue));
    }, [searchValue]);
    useLayoutEffect(() => {
        const input = inputRef.current && inputRef.current;
        if (isSearchOpen && input && !inputValue) {
            input.focus();
            setInputFocus(true);
        }
    }, [isSearchOpen, inputValue]);

    const toggleInputFocus = () => setInputFocus(!inputFocus);

    const handleCloseSearch = () => {
        setInputValue('');
        setIsSearchOpen(false);
        setInputFocus(false);
    };

    const handleClickSearch = () => {
        if (isSearchOpen && inputValue) {
            navigate(`/${searchUrl}?query=${inputValue}`);
        } else {
            setIsSearchOpen(!isSearchOpen);
            handleDropdown(false);
            setInputFocus(false);
        }
    };

    const stateSelectIcon = isDropdownOpen ? ComponentState.selected : ComponentState.normal;
    const stateSearchIcon = inputFocus ? ComponentState.hover : ComponentState.normal;
    return (
        <StyledMobileCategoryHeader>
            <div>
                <div className="header">
                    <StyledCategorySelect
                        isSearchOpen={isSearchOpen}
                        isDropdownOpen={isDropdownOpen}
                        onClick={() => handleDropdown(!isDropdownOpen)}
                    >
                        <div className="title">{title}</div>
                        <Icon iconName={GlobalIconName.downChevron16} state={stateSelectIcon} />
                    </StyledCategorySelect>
                    <StyledCategorySelectSearch isOpen={isSearchOpen} className={isSearchOpen ? 'open' : ''}>
                        <button
                            type="button"
                            data-testid="search-button"
                            className="search-button"
                            aria-label="search button"
                            onClick={handleClickSearch}
                        >
                            <Icon iconName={GlobalIconName.search24} state={stateSearchIcon} />
                        </button>
                        {isSearchOpen && (
                            <>
                                <input
                                    ref={inputRef}
                                    type="search"
                                    value={inputValue}
                                    placeholder="Search here"
                                    name="search"
                                    id="search"
                                    data-testid="search-input"
                                    aria-label="search input"
                                    onChange={e => setInputValue(e.target.value)}
                                    onFocus={toggleInputFocus}
                                    onBlur={toggleInputFocus}
                                    onKeyDown={e => {
                                        if (e.key === 'Enter') handleClickSearch();
                                    }}
                                />
                                <button
                                    type="button"
                                    className="close-button"
                                    aria-label="close search"
                                    onClick={handleCloseSearch}
                                >
                                    <Icon iconName={GlobalIconName.close16} state={ComponentState.normal} />
                                </button>
                            </>
                        )}
                    </StyledCategorySelectSearch>
                </div>
                <div className="divider" />
            </div>
        </StyledMobileCategoryHeader>
    );
}

export function MobileCategoryBody({
    title,
    categoryList,
    urlActive,
    isDropdownOpen = false,
    handleCloseDropdown,
}: MobileCategoryBodyProps) {
    const containerRef = useRef<HTMLDivElement>(null);
    const contentRef = useRef<HTMLDivElement>(null);
    const [hasBorderBottom, setHasBorderBottom] = useState<boolean>(false);

    useEffect(() => {
        const containerMenuRef = containerRef?.current || null;

        const handleHeaderScroll = () => {
            if (!contentRef.current) return;
            const { top } = contentRef.current.getBoundingClientRect();
            setHasBorderBottom(!!(top < 59.99));
        };

        if (containerMenuRef && isDropdownOpen) {
            containerMenuRef.addEventListener('scroll', handleHeaderScroll);
            const topPosition = document.body.getBoundingClientRect().top;
            document.body.style.top = `${topPosition}px`;
            document.body.style.position = 'fixed';
        } else {
            document.body.style.position = '';
            const positionString = document.body.style.top.replace('px', '');
            document.body.scrollTop += Math.abs(Number(positionString));
            document.documentElement.scrollTop += Math.abs(Number(positionString));
            document.body.style.top = '';
        }

        return () => {
            containerMenuRef?.removeEventListener('scroll', handleHeaderScroll);
        };
    }, [hasBorderBottom, isDropdownOpen]);

    if (!categoryList?.length || !isDropdownOpen) return null;

    return (
        <>
            <StyledMobileCategoryBodyHeader className={hasBorderBottom ? 'border-bottom' : ''}>
                <h1>{title}</h1>
                <div>
                    <button
                        type="button"
                        className="close-button"
                        aria-label="close search"
                        onClick={() => handleCloseDropdown()}
                    >
                        <Icon iconName={GlobalIconName.close16} state={ComponentState.hover} />
                    </button>
                </div>
            </StyledMobileCategoryBodyHeader>
            <StyledCategoryDropdownShape isDropdownOpen={isDropdownOpen} ref={containerRef}>
                <div className="content" ref={contentRef}>
                    <ul>
                        {categoryList.map(item => {
                            const isLinkActive = urlActive === item?.uri;

                            if (item?.wpChildren?.nodes?.length) {
                                return (
                                    <MobileCategorySubMenu
                                        key={item.id}
                                        title={item.name!}
                                        categoryList={item.wpChildren?.nodes!}
                                        urlActive={urlActive}
                                    />
                                );
                            }
                            return (
                                <ListItemAnchor
                                    key={item?.id}
                                    title={item?.name!}
                                    url={item?.uri!}
                                    className={isLinkActive ? 'active' : ''}
                                />
                            );
                        })}
                    </ul>
                </div>
            </StyledCategoryDropdownShape>
        </>
    );
}

export default function MobileCategoryMenu({
    title = 'Select Category',
    categoryList,
    searchUrl,
    searchValue = '',
    urlActive = '',
    hasTopBanner = false,
    stick = true,
}: MobileCategoryMenuProps) {
    const [isDropdownOpen, setIsDropdownOpen] = useState<boolean>(false);
    const { isStick, divRef } = useStick();

    const renderHeader = () => (
        <MobileCategoryHeader
            title={title}
            searchUrl={searchUrl}
            searchValue={searchValue}
            isDropdownOpen={isDropdownOpen}
            handleDropdown={setIsDropdownOpen}
        />
    );

    return (
        <div ref={divRef}>
            <StyledMobileCategoryStick hasTopBanner={hasTopBanner} showStick={stick} stick={isStick}>
                {renderHeader()}
            </StyledMobileCategoryStick>
            <StyledMobileCategoryMenu hasTopBanner={hasTopBanner}>{renderHeader()}</StyledMobileCategoryMenu>
            <MobileCategoryBody
                title={title}
                categoryList={categoryList}
                urlActive={urlActive}
                isDropdownOpen={isDropdownOpen}
                handleCloseDropdown={() => setIsDropdownOpen(false)}
            />
        </div>
    );
}
