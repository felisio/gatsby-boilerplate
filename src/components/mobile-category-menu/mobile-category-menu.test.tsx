import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import React from 'react';
import { MobileCategoryHeaderProps, MobileCategoryHeader, MobileCategoryBody } from './mobile-category-menu';

import categoryList from '../../../__mocks__/category-menu';
import { WpCategory } from '../../../graphql-types';

const createTestProps = (props: MobileCategoryHeaderProps) => ({
    ...props,
});

const title = 'Select Category';
const props = createTestProps({
    title,
    handleDropdown: () => false,
});

test('should verify MobileCategoryMenu Component snapshot', async () => {
    const tree = renderer.create(
        <MobileCategoryBody
            title={title}
            categoryList={categoryList as WpCategory[]}
            isDropdownOpen
            handleCloseDropdown={() => false}
        />,
    );

    expect(tree).toMatchSnapshot();
});

test('should verify that MobileCategoryMenu Component shows input search', () => {
    render(<MobileCategoryHeader {...props} />);

    const button = screen.getByTestId('search-button');
    expect(button).toBeInTheDocument();
    fireEvent.click(button);
    expect(button.parentElement).toHaveClass('open');
});

test('should verify form submit', () => {
    render(<MobileCategoryHeader {...props} />);

    const button = screen.getByTestId('search-button');
    fireEvent.click(button);
    const input = screen.getByTestId('search-input');
    expect(input).toBeInTheDocument();
});

test('should verify that input search has value', () => {
    const searchValue = 'Quantic search';
    render(<MobileCategoryHeader {...props} searchValue={searchValue} />);

    const input = screen.getByTestId('search-input') as HTMLInputElement;
    expect(input.value).toBe(searchValue);
});

test('should verify active link', () => {
    const urlActive = '/category/entrepreneurship/';
    render(
        <MobileCategoryBody
            title={title}
            categoryList={categoryList as WpCategory[]}
            isDropdownOpen
            handleCloseDropdown={() => false}
            urlActive={urlActive}
        />,
    );

    const listItem = screen.getByText('Entrepreneurship');
    expect(listItem.parentElement).toHaveClass('active');
});
