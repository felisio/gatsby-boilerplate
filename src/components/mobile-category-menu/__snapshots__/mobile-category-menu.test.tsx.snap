// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should verify MobileCategoryMenu Component snapshot 1`] = `
Array [
  .c0 {
  position: fixed;
  top: 0;
  left: 0;
  z-index: 20;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  height: 60px;
  padding: 1rem 1.5rem 1.3rem 2.4rem;
  overflow-y: hidden;
  background-color: var(--white);
}

.c0.border-bottom::after {
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 1px;
  content: '';
  background-color: rgba(121,137,159,0.8);
  opacity: 0.15;
}

.c0 h1 {
  padding: 0;
  margin: 0;
  font-size: 1.8rem;
  font-weight: 500;
  font-stretch: normal;
  color: var(--wild-watermelon);
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c0 button {
  position: relative;
  cursor: pointer;
  background-color: transparent;
  border: none;
  outline: none;
}

.c0 button:focus {
  border: none;
  outline: none;
}

.c1.icon-12 {
  width: 1.2rem;
  height: 1.2rem;
}

.c1.icon-16 {
  width: 1.6rem;
  height: 1.6rem;
}

.c1.icon-18 {
  width: 1.8rem;
  height: 1.8rem;
}

.c1.icon-24 {
  width: 2.4rem;
  height: 2.4rem;
}

.c1.icon-32 {
  width: 3.2rem;
  height: 3.2rem;
}

.c1.icon-128 {
  width: 12.8rem;
  height: 3.8rem;
}

@media (orientation:landscape) {

}

<header
    className="c0"
  >
    <h1>
      Select Category
    </h1>
    <div>
      <button
        aria-label="close search"
        className="close-button"
        onClick={[Function]}
        type="button"
      >
        <img
          alt="close"
          className="c1 icon-16"
          src="test-file-stub"
        />
      </button>
    </div>
  </header>,
  .c0 {
  position: fixed;
  top: 60px;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 19;
  display: block;
  height: 100vh;
  overflow-x: hidden;
  overflow-y: auto;
  pointer-events: auto;
  background-color: var(--white);
  background-clip: padding-box;
  opacity: 1;
  -webkit-transition: 0.2s ease-in-out;
  transition: 0.2s ease-in-out;
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
  --padding-bottom: calc(100% - 22em);
}

.c0 .content {
  padding: 0 2.4rem var(--padding-bottom) 2.4rem;
}

.c0 ul {
  padding: 0;
  margin: 0;
  overflow: auto;
  list-style: none;
}

.c0 ul li.submenu {
  --border-color: rgba(121,137,159,0.1);
  border-top: 1px solid var(--border-color);
  border-bottom: 1px solid var(--border-color);
}

.c0 ul li.submenu:last-child {
  border-bottom: none;
}

.c0 ul li.submenu:first-child {
  border-top: none;
}

.c0 ul li a {
  display: block;
}

.c0 ul li a,
.c0 ul li button {
  padding: 0.4rem 0;
  font-size: 1.5rem;
  font-weight: 500;
  line-height: 2;
  color: var(--shuttle-grey);
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0.23px;
  -moz-letter-spacing: 0.23px;
  -ms-letter-spacing: 0.23px;
  letter-spacing: 0.23px;
  -webkit-transition: 0.2s ease;
  transition: 0.2s ease;
}

.c0 ul li.active a {
  color: var(--wild-watermelon);
}

.c2 {
  --border-color: rgba(121,137,159,0.1);
  margin-top: 0.4rem;
  border-top: 1px solid var(--border-color);
  border-bottom: 1px solid var(--border-color);
}

.c2:last-child {
  border-bottom: none;
}

.c2:first-child {
  border-top: none;
}

.c2 button {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  padding: 0;
  margin: 0;
  text-align: start;
  cursor: pointer;
  background-color: transparent;
  border: none;
  outline: none;
}

.c2 button:focus {
  border: none;
  outline: none;
}

.c2 button img {
  position: relative;
  top: 9px;
  -webkit-transition: all 0.5s ease-in-out;
  transition: all 0.5s ease-in-out;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c2 ul {
  display: none;
  width: 100%;
  opacity: 0;
  -webkit-transition: 0.2s ease-in-out;
  transition: 0.2s ease-in-out;
  -webkit-transform: translateY(-8px);
  -ms-transform: translateY(-8px);
  transform: translateY(-8px);
}

.c2 ul li {
  padding-left: 1.6rem;
}

.c2 ul li a {
  display: block;
}

.c2 ul li a,
.c2 ul li button {
  padding: 0.4rem 0;
  font-size: 1.5rem;
  font-weight: 500;
  line-height: 2;
  color: var(--shuttle-grey);
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0.23px;
  -moz-letter-spacing: 0.23px;
  -ms-letter-spacing: 0.23px;
  letter-spacing: 0.23px;
  -webkit-transition: 0.2s ease;
  transition: 0.2s ease;
}

.c2 ul li.active a {
  color: var(--wild-watermelon);
}

.c3.icon-12 {
  width: 1.2rem;
  height: 1.2rem;
}

.c3.icon-16 {
  width: 1.6rem;
  height: 1.6rem;
}

.c3.icon-18 {
  width: 1.8rem;
  height: 1.8rem;
}

.c3.icon-24 {
  width: 2.4rem;
  height: 2.4rem;
}

.c3.icon-32 {
  width: 3.2rem;
  height: 3.2rem;
}

.c3.icon-128 {
  width: 12.8rem;
  height: 3.8rem;
}

.c1 {
  padding: 0;
  margin: 0;
}

@media (orientation:landscape) {
  .c0 .content {
    --padding-bottom: calc(50% - 22em);
  }
}

<div
    className="c0"
  >
    <div
      className="content"
    >
      <ul>
        <li
          className="c1"
        >
          <a
            href="/category/home/"
          >
            Home
          </a>
        </li>
        <li
          className="c1"
        >
          <a
            href="/category/entrepreneurship/"
          >
            Entrepreneurship
          </a>
        </li>
        <li
          className="c1"
        >
          <a
            href="/category/accounting/"
          >
            Accounting
          </a>
        </li>
        <li
          className="c1"
        >
          <a
            href="/category/finance/"
          >
            Finance
          </a>
        </li>
        <li
          className="c1"
        >
          <a
            href="/category/changethecourse/"
          >
            #ChangeTheCourse
          </a>
        </li>
        <li
          className="c1"
        >
          <a
            href="/category/alumni/"
          >
            Alumni
          </a>
        </li>
        <li
          className="c1"
        >
          <a
            href="/category/careers/"
          >
            Careers
          </a>
        </li>
        <li
          className="c1"
        >
          <a
            href="/category/companies-universities/"
          >
            Companies & Universities
          </a>
        </li>
        <li
          className="c2"
        >
          <button
            onClick={[Function]}
            type="button"
          >
            Fields of Study
            <img
              alt="downChevron"
              className="c3 icon-16"
              src="test-file-stub"
            />
          </button>
          <ul>
            <li
              className="c1"
            >
              <a
                href="/category/fields-of-study/accounting/"
              >
                Accounting
              </a>
            </li>
            <li
              className="c1"
            >
              <a
                href="/category/fields-of-study/data-analytics/"
              >
                Data Analytics
              </a>
            </li>
            <li
              className="c1"
            >
              <a
                href="/category/fields-of-study/finance/"
              >
                Finance
              </a>
            </li>
            <li
              className="c1"
            >
              <a
                href="/category/fields-of-study/strategy/"
              >
                Strategy
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </div>
  </div>,
]
`;
